src/index.d.ts

```diff
@@ -18,7 +18,7 @@ type WithThemeFactory<Theme> = <
 ) => React.ComponentType<OuterProps>;
 
 interface ThemeProviderProps<Theme> {
-    theme: Theme | ((outerTheme: object) => Theme),
```

Comment:
The problem with changing the definition of ThemeProviderProps not to allow function is a problem.
It makes sense that your App component only receives a Theme Object, but a function should still be allowed for the actual theme prop of the ThemeProvider.

Reply:
Yes, I would like to support both also, but apperently using a object is currently not possible, even with this signature?

Reply:
Is it possible to pass an object to `ThemeProvider` as the `theme` prop or is this also not working?

Reply:
You're right. I had some props with some conflicting typings!